lib/rack/server_status.rb

```diff
@@ -19,18 +19,13 @@ def initialize(app, options = {})
     def call(env)
       set_state!('A', env)
 
-      @clean_proc = Remover.new
-      ObjectSpace.define_finalizer(self, @clean_proc)
-
-      if @path && env['PATH_INFO'] == @path
-        res = handle_server_status(env)
-        set_state!('_')
-        return res
+      if env['PATH_INFO'] == @path
```

Comment:
`@path` is set to default `/server-status` when `option[:path]` is falsey. Therefore, I removed unnecessary check for `@path`.